### PR TITLE
[20.09] nixUnstable: 2.4pre20210326_dd77f71 -> 2.4pre20210503_6d2553a; hydra-unstable: 2021-03-29 -> 2021-05-03

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

Backport PR for https://github.com/NixOS/nixpkgs/pull/121263, https://github.com/NixOS/nixpkgs/pull/121731

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by cole-h — APPROVED** [review]

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -3842,6 +3856,33 @@ let
...
+  CryptPassphrase = buildPerlPackage {
+    pname = "Crypt-Passphrase";
+    version = "0.003";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/L/LE/LEONT/Crypt-Passphrase-0.003.tar.gz";
+      sha256 = "685aa090f8179a86d6896212ccf8ccfde7a79cce857199bb14e2277a10d240ad";
+    };
+    meta = {
+      description = "A module for managing passwords in a cryptographically agile manner";
+      license = with lib.licenses; [ artistic1 gpl1Plus ];
```
Comment (cole-h) [inline pkgs/top-level/perl-packages.nix:3868]: Looks like release-20.09 doesn't have 276f91ec394f, so these commits will need to be modified by hand to reference `stdenv.lib` (or cherry-pick that commit...)

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -3842,6 +3856,33 @@ let
...
+    pname = "Crypt-Passphrase-Argon2";
+    version = "0.002";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/L/LE/LEONT/Crypt-Passphrase-Argon2-0.002.tar.gz";
+      sha256 = "3906ff81697d13804ee21bd5ab78ffb1c4408b4822ce020e92ecf4737ba1f3a8";
+    };
+    propagatedBuildInputs = with perlPackages; [ CryptArgon2 CryptPassphrase ];
+    meta = {
+      description = "An Argon2 encoder for Crypt::Passphrase";
+      license = with lib.licenses; [ artistic1 gpl1Plus ];
```
Comment (cole-h) [inline pkgs/top-level/perl-packages.nix:3882]: And here...

File: pkgs/top-level/perl-packages.nix [hunk 3]
```diff
@@ -17737,6 +17778,19 @@ let
...
+  StringCompareConstantTime = buildPerlPackage {
+    pname = "String-Compare-ConstantTime";
+    version = "0.321";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/F/FR/FRACTAL/String-Compare-ConstantTime-0.321.tar.gz";
+      sha256 = "0b26ba2b121d8004425d4485d1d46f59001c83763aa26624dff6220d7735d7f7";
+    };
+    meta = {
+      description = "Timing side-channel protected string compare";
+      license = with lib.licenses; [ artistic1 gpl1Plus ];
```
Comment (cole-h) [inline pkgs/top-level/perl-packages.nix:17790]: And here.

File: pkgs/top-level/perl-packages.nix (hunk 1 above)
Comment (Ma27) [reply pkgs/top-level/perl-packages.nix:3868]: Hmm it's kinda weird that my `nix-build` only caught this for one package oO
Will fix it :)

**Review by thufschmitt — APPROVED** [review]
The [ofborg failure](https://logs.nix.ci/?attempt_id=a117e723-a70d-435c-9490-537aa90486d4&key=nixos%2Fnixpkgs.121739) on aarch64-linux is a bit strange, but nixUnstable was already red on hydra, so… 🤷‍♂️

Apart from that 👍🏻

**Pull request merged**